### PR TITLE
fix(tracing): do not serialise out-of-process Buffers

### PR DESCRIPTION
**Pull request by mxschmitt**

### CODE REVIEW

File: packages/playwright-core/src/server/trace/recorder/tracing.ts [hunk 1]
```diff
@@ -496,10 +497,12 @@ export class Tracing extends SdkObject implements InstrumentationListener, Snaps
...
   if (Array.isArray(object))
     return object.map(o => visitTraceEvent(o, sha1s));
+  if (object instanceof Dispatcher)
+    return `<${(object as Dispatcher<any, any, any>)._type}>`;
   if (object instanceof Buffer)
-    return undefined;
+    return `<Buffer>`;
   if (object instanceof Date)
-    return object;
+    return `<Date:${object.toJSON()}>`;
```
Comment (pavelfeldman) [inline packages/playwright-core/src/server/trace/recorder/tracing.ts:505]: Why did this change? Sounds like we had better traces with it as is?

File: tests/library/tracing.spec.ts [hunk 2]
```diff
@@ -109,9 +109,7 @@ test('should not collect snapshots by default', async ({ context, page, server }
   expect(events.some(e => e.type === 'resource-snapshot')).toBeFalsy();
 });
 
-test('should not include buffers in the trace', async ({ context, page, server, mode }, testInfo) => {
-  test.skip(mode !== 'default', 'no buffers with remote connections');
```
Comment (mxschmitt) [inline tests/library/tracing.spec.ts:113]: Note: Removing the skip covers the bug with a test.

File: packages/playwright-core/src/server/trace/recorder/tracing.ts [hunk 3]
```diff
@@ -496,8 +497,10 @@ export class Tracing extends SdkObject implements InstrumentationListener, Snaps
 function visitTraceEvent(object: any, sha1s: Set<string>): any {
   if (Array.isArray(object))
     return object.map(o => visitTraceEvent(o, sha1s));
+  if (object instanceof Dispatcher)
```
Comment (pavelfeldman) [inline packages/playwright-core/src/server/trace/recorder/tracing.ts:500]: when does this happen?

Comment (mxschmitt) [reply packages/playwright-core/src/server/trace/recorder/tracing.ts:500]: When we call `_appendTraceEvent` (after) for `newPage`. This returns `{ result: PageDispatcher }`.

**Review by pavelfeldman — APPROVED** [review]

**Pull request merged**